### PR TITLE
Minificpp 897

### PR DESCRIPTION
**Pull request by am-c-p-p**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: CMakeSettings.json [hunk 1]
```diff
@@ -104,6 +104,10 @@
       "cmakeCommandArgs": "",
       "buildCommandArgs": "-v -j8",
       "variables": [
+        {
+          "name": "ENABLE_LIBMODBUS",
+          "value": "ON"
```
Comment (arpadboda) [inline CMakeSettings.json]: This seems to be out of sync with the option in cmakelists.txt, where it defaults to off.

File: extensions/libmodbus/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+include(${CMAKE_SOURCE_DIR}/extensions/ExtensionHeader.txt)
+
+include(ExternalProject)
+
+ExternalProject_Add(
+  libmodbus_external
+  PREFIX ${LIBMODBUS_ROOT}
+  GIT_REPOSITORY "https://github.com/stephane/libmodbus.git"
```
Comment (arpadboda) [inline extensions/libmodbus/CMakeLists.txt]: Can you specify a tag or at least a commit here to make sure our builds doesn't get accidentally broken by changes in this repo?

File: CMakeSettings.json (hunk 1 above)
Comment (phrocker) [reply CMakeSettings.json]: This is a different environment and specifies a build, so he is enabling it here.

File: extensions/libmodbus/Libmodbus.vcxproj [hunk 3]
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (phrocker) [inline extensions/libmodbus/Libmodbus.vcxproj]: is this a derivative of what is in libmodbus or is this one you created entirely on your own? What is the difference between this one and the one that is in their repo?

Comment (am-c-p-p) [reply extensions/libmodbus/Libmodbus.vcxproj]: It is not a derivative, I created it.

**Comment by phrocker** [comment]
@am-c-p-p with your last comment I'm good with this; however, I don't see any thing being compiled with this.  I also appreciate @arpadboda 's comment. Can you add some readme to this extension as to why these files are needed for this extension? I think it's fair that the purview of an extension allows it to use a process, but a readme of why would be helpful.

Do you intend to submit a separate PR for follow on code to this or will you update this one?

**Comment by am-c-p-p** [comment]
@phrocker Yes, it will be separate PR.

**Comment by szaszm** [comment]
Closing this because of inactivity

**Pull request closed without merging**